### ISSUE TITLE
Do we even need SD 1.5 for image restoration? Why not use video restoration models instead of SD 1.5 for image restoration mode?

### ISSUE THREAD

**Issue by TonySSS100** [Kruk2/jasna#204]
Personally I find the usage of SD 1.5 model for image restoration completely worthless, as its known to inpaint any censored part and reconstruct it rather than restoring it like with video models. So I say to just use the video restoration models for image restoration mode too and completely ditch SD 1.5 and other versions altogether.

My current workaround to decensor images in Jasna (same counts for Lada too as that's where I first started implementing it long ago) is encoding an image into a video with multiple frames (Ctrl+C and V multiple times as needed) in Avidemux with a specific codec called Mpeg4 AVC (x264) with these settings in order to make the image as close as possible to original source, since other encoders just screw with color tonality: 

<img width="607" height="702" alt="Image" src="https://github.com/user-attachments/assets/7fe81474-181e-4cf4-9d28-7e65e23ab232" />

With JPG format images it always works fine, however with many PNG ones instead it always gives me an error and won't export it at all, not to mention having green borders on a PNG image when imported in Avidemux for some reason, even when I try to export a PNG to JPG, encode it as video with Mpeg4 AVC and decensor it in Jasna, it always gives me divided by zero error.

Lastly both detection models can struggle with static images sometimes, so I would like to have a manual masking option too for images so both video restoration models can brute super resolute the area you manually masked so it can always decensor images properly.

**Comment by teijiIshida** [comment]
Having another option is good but definitely do not "ditch SD15". Some of us just needs realistic-looking pictures to trick the brain and SD15 delivers (more so than current non-inpaint model).

**Comment by TonySSS100** [comment]
> Having another option is good but definitely do not "ditch SD15". Some of us just needs realistic-looking pictures to trick the brain and SD15 delivers (more so than current non-inpaint model).

I do agree that SD 1.5 delivers sometimes, but I see using that model more for images that are censored with bars or other solid shapes that completely cover genitals instead of pixellation.

**Comment by Kruk2** [comment]
### tl;dr
If I can't make good restorations on still images then there is no point in even trying realistic restoration on videos.

### Longer version
Good (and currently released sd 1.5 model is not good) spatial model is handy to have, valuable experiment and some people actually wanted to restore images so I decided to give them this pretrained model which is one step in my video diffusion pipeline.

If I would have cluster of H100 and millions of data I would jump straight to video diffusion model but I don't so starting with single-frame only model is my attempt. Later you can add temporal attention, 3d convolutions, flow-wrap (this is actually what lada uses) to see what will happen and if you will get coherent video restoration. **[Paper](https://arxiv.org/abs/2304.08818)**\
Key sentence: **We first pre-train an LDM on images only; then, we turn the image generator into a video generator by introducing a temporal dimension to the latent space**

Lastly, I'm not using sd 1.5 in a way you think. True, it invents new texture but its heavily guided by the blocky mosaic under it. Released model uses Stochastic Differential Editing to use mosaic as guidance [**Paper**](https://arxiv.org/abs/2108.01073)
You can actually see it if you lower SDEdit parameter to 0.1-0.2. It will blur mosaic, then add more details with each step.

I'm surprised about detection models because they're trained only on still images, they're not temporal and shouldn't be in my opinion.
Neither ladaapp nor me trained detection on many frames at once.
I thought about manual masking but I don't think its worth my time to implement it atm

**Comment by TonySSS100** [comment]
> Good (and currently released sd 1.5 model is not good) spatial model is handy to have, valuable experiment and some people actually wanted to restore images so I decided to give them this pretrained model which is one step in my video diffusion pipeline.
> 
> If I would have cluster of H100 and millions of data I would jump straight to video diffusion model but I don't so starting with single-frame only model is my attempt. Later you can add temporal attention, 3d convolutions, flow-wrap (this is actually what lada uses) to see what will happen and if you will get coherent video restoration. **[Paper](https://arxiv.org/abs/2304.08818)** Key sentence: **We first pre-train an LDM on images only; then, we turn the image generator into a video generator by introducing a temporal dimension to the latent space**
> 
> Lastly, I'm not using sd 1.5 in a way you think. True, it invents new texture but its heavily guided by the blocky mosaic under it. Released model uses Stochastic Differential Editing to use mosaic as guidance [**Paper**](https://arxiv.org/abs/2108.01073) You can actually see it if you lower SDEdit parameter to 0.1-0.2. It will blur mosaic, then add more details with each step.
> 
> I'm surprised about detection models because they're trained only on still images, they're not temporal and shouldn't be in my opinion. Neither ladaapp nor me trained detection on many frames at once. I thought about manual masking but I don't think its worth my time to implement it atm

I do agree that a spatial model is the best thing to have at the moment for restoring videos and images with pixellation, and yes training one does require a heftly powerful server that costs a lot and takes months to years for a complete training, no doubts about that.

As for detection models, the current workaround I found is to simply crop the pixelated area and decensor that in Jasna (hoping that ffmpeg won't throw out an error due to a size requirement of sorts), but still having an in app manual masking model would be still nice to have IMO.

**Comment by cbl19961214-sudo** [comment]
我认为目前视频恢复模型能做到程度就是还原大体结构，根据马赛克的不同大小和模糊程度，细节可能只还原了30~50%不等，与其死磕下去，不如用sd模型在结构基础上重绘细节，增加清晰度，只要足够自然，去除掉了马赛克的不美观，达到欺骗大脑的效果，无人在意是否真正的“还原”。
